### PR TITLE
Add pytest workflow for CI

### PR DESCRIPTION
**Pull request by maxulysse**
Following @Emiller88 lead on improving CI

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
@Emiller88 I see what you meant by this where the real magic starts...
:exploding_head:

File: tests/config/test_data.config [hunk 1]
```diff
@@ -0,0 +1,16 @@
+def test_data_dir = "${launchDir}/tests/data/"
+def nf_core_modules_data = "https://raw.githubusercontent.com/nf-core/modules/master/tests/data/"
```
Comment (Emiller88) [inline tests/config/test_data.config:2]:
```suggestion
def nf_core_modules_data = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/"
```

Forgot this got updated :grimacing:

**Review by FriederikeHanssen — COMMENTED** [review]
Wow, this is awesome 🤩 ❤️ . The linting i not passing yet:

```
[✗] 21 Tests Failed                                                          │
├──────────────────────────────────────────────────────────────────────────────┤
│ files_exist: File not found: lib/nfcore_external_java_deps.jar               │
│ files_exist: File not found: lib/NfcoreSchema.groovy                         │
│ nextflow_config: Config variable not found: params.show_hidden_params        │
│ nextflow_config: Config variable not found: params.schema_ignore_params      │
│ nextflow_config: Config variable (incorrectly) found: params.name            │
│ files_unchanged: CODE_OF_CONDUCT.md does not match the template              │
│ files_unchanged: .github/.dockstore.yml does not match the template          │
│ files_unchanged: .github/CONTRIBUTING.md does not match the template         │
│ files_unchanged: .github/ISSUE_TEMPLATE/bug_report.md does not match the te… │
│ files_unchanged: .github/ISSUE_TEMPLATE/feature_request.md does not match t… │
│ files_unchanged: .github/markdownlint.yml does not match the template        │
│ files_unchanged: .github/PULL_REQUEST_TEMPLATE.md does not match the templa… │
│ files_unchanged: .github/workflows/branch.yml does not match the template    │
│ files_unchanged: .github/workflows/linting.yml does not match the template   │
│ files_unchanged: assets/nf-core-sarek_logo.png does not match the template   │
│ files_unchanged: bin/markdown_to_html.py does not match the template         │
│ files_unchanged: docs/images/nf-core-sarek_logo.png does not match the temp… │
│ files_unchanged: .gitignore does not match the template                      │
│ files_unchanged: assets/multiqc_config.yaml does not match the template      │
│ actions_ci: '.github/workflows/ci.yml' is not triggered on expected events   │
│ actions_ci: Minimum NF version in '.github/workflows/ci.yml' different to p… │
```

Did you expect that? 

And one other idea: maybe add a short readme on how to add new tests for new developers, or is it the exact same as with the modules-repo?

**Comment by maxulysse** [comment]
Yeah, nf-core lint would be another PR.
Same as for modules. but yeah a README somewhere might be nice

**Review by FriederikeHanssen — APPROVED** [review]
OK great. Then maybe let's add a README in a seperate PR, like a short Step-by-step or so.

**Comment by maxulysse** [comment]
> OK great. Then maybe let's add a README in a seperate PR, like a short Step-by-step or so.

I'm guessing a step by step should even go in another repo.
It's not sarek specific

**Pull request merged**